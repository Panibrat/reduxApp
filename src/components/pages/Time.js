import React from 'react';
import moment from 'moment';

export default class Time extends React.Component {
  constructor(props) {
    super(props);
    this.state = { time: moment().format("dddd, MMMM Do YYYY, h:mm:ss a") };
  }

  tick() {
    this.setState(() => ({
      time: moment().format("dddd, MMMM Do YYYY, h:mm:ss a")
    }));
  }

  componentDidMount() {
    this.interval = setInterval(() => this.tick(), 1000);
  }

  componentWillUnmount() {
  clearInterval(this.interval);
}

render() {
  return (
    <div>
      <h1>Time:</h1>
      <h4>{this.state.time}</h4>
    </div>
  );
}
}
