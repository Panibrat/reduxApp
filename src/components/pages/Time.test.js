import React from 'react';
import {shallow} from 'enzyme';
import Time from './Time';

describe('Time Rendering', () => {

  test('should renderTime correctly ', () => {
    const wrapper = shallow(<Time/>);
    expect(wrapper).toMatchSnapshot();

  });

});
