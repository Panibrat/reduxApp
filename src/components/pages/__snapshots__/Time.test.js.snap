// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Time Rendering should renderTime correctly  1`] = `
<div>
  <h1>
    Time:
  </h1>
  <h4>
    Thursday, January 1st 1970, 2:00:00 am
  </h4>
</div>
`;
